views/pulsar/v2/helpers/html.html.twig

```diff
@@ -1010,23 +1010,43 @@ Option | Type   | Description
 -----  | ------ | --------------------------------------------------------------
 class  | string | CSS classes, space separated
 id     | string | A unique identifier, if required
+message| string | Optional text string to be shown in block style loading elements
+type   | string | `block` or `inline` (default), block will be centered horizontally
 data-* | string | Data attributes, eg: `'data-foo': 'bar'`
 
 #}
 {% macro loading(options) %}
 {% spaceless %}
     {% import '@pulsar/pulsar/v2/helpers/util.html.twig' as util %}
 
-    <span{{
-        attributes(options
-            |exclude('label')
-            |defaults({
-                'class': 'loading'
-            })
-        )
-    }}>
-        <i>{{- options.label|default('Loading...') -}}</i>
-    </span>
+    {% set type = 'inline' %}
+    {% if options.type is defined and options.type == 'block' %}
+        {% set type = 'block' %}
+    {% endif %}
+
+    {% if type == 'block' %}
+    <div class="loading-block">
+    {% endif %}
+
+        <span{{
+            attributes(options
+                |exclude('label')
+                |defaults({
+                    'class': 'loading',
+                    'role': 'status'
```

Comment:
`role="status"` is redundant here, text would only be announced if the text was added/changed after page load.